R/pretty_lollipop_plot.R

```diff
@@ -2,58 +2,118 @@
-#'                      gene = "MYC",
-#'                      plot_title = "Mutation data for MYC",
-#'                      plot_theme = "nature2")
+#' lolipop_result <- get_lolipop_by_gene_symbol(maf_data, "MYC")
+#' print(lolipop_result)
 #'
-pretty_lollipop_plot = function(maf_df,
-                                gene,
-                                plot_title,
-                                plot_theme = "cbioportal",
-                                out_name = paste0("my_lollipop_plot_", gene)){
-  if(missing(gene)){
-    stop("Plese provide a gene...")
+get_lolipop_by_gene_symbol <- function(maf_data, 
+                                       gene_symbol = NULL) {
+  # Read the MAF object if it's not already in MAF format
+  if (!inherits(maf_data, "MAF")) {
+    maf_data <- read.maf(maf_data)
   }
+  
+  maf_data <- maf_data@data 
+  # Filter for the specific gene 
+  gene_df <- maf_data %>% 
+    filter(Hugo_Symbol == gene_symbol) %>%
+    mutate(AA = as.numeric(gsub("[^0-9]+", "", gsub("([0-9]+).*", "\\1", HGVSp_Short)))) %>%
+    arrange(AA)
 
-  if(missing(plot_title)){
-    plot_title = gene
+  gene_counts <- gene_df %>% 
+    group_by(AA, Start_Position, 
+             End_Position, 
+             Variant_Classification, 
+             Reference_Allele, 
+             Tumor_Seq_Allele2) %>%
+    arrange(AA) %>%
+    summarise(mutation_count = n()) 
+
+  protein_domain_subset <- subset(protein_domains, HGNC == gene_symbol)
+  if (!is.data.frame(protein_domain_subset)) {
+    protein_domain_subset <- as.data.frame(protein_domain_subset)
   }
+   
+  domain_data <- protein_domain_subset %>% data.frame(start.points = protein_domain_subset$Start,
+                                                      end.points = protein_domain_subset$End,
+                                                      text.label = protein_domain_subset$Label,
+                                                      color = protein_domain_subset$Label)
+
+  domain_data$text.position <- (domain_data$start.points + domain_data$end.points) / 2
+
+  # Determine the x-axis range
+  x_max <- max(max(domain_data$end.points), 
+               max(gene_counts$AA, na.rm = TRUE))
+  x_min <- 0
+
+  colors_manual <- c(setNames(protein_domain_subset$Label, protein_domain_subset$Label), 
+                     "Frame_Shift_Del" = "#e50505", 
+                     "Frame_Shift_Ins" = "#df9307" ,
+                     "In_Frame_Del" = "#f6f60c",
+                     "In_Frame_Ins" = "#04e504",
+                     "Missense_Mutation" = "#9e12f6", 
+                     "Nonsense_Mutation" = "#06ebeb",
+                     "Nonstop_Mutation" = "#0505df", 
+                     "Splice_Site" = "#035703",
+                     "Translation_Start_Site" = "#f3899b",
+                     "RNA" = "#146deb")
+  
+  Somatic_Mutation_Statistic_Numerator <- maf_data %>%
+    filter(Hugo_Symbol == gene_symbol) %>% 
+    filter(!(Variant_Classification %in% c("Intron",
```

Comment:
Here we can also use global variables: `GAMBLR.helpers::coding_class` to filter the maf data for coding-only mutations